sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/statefulOperators.scala

```diff
@@ -132,6 +135,16 @@ trait StateStoreWriter extends StatefulOperator { self: SparkPlan =>
     }.toMap
   }
 
+  protected def applyRemovingRowsOlderThanWatermark(
+      iter: Iterator[InternalRow],
+      predicate: BasePredicate): Iterator[InternalRow] = {
+    iter.filter { row =>
+      val filteredIn = !predicate.eval(row)
```

Comment:
nit: little confusing of the variable naming here, maybe use a name related with watermark? `inWatermark`?

Reply:
Maybe it's clearer if we apply `filterNot` - then the variable name can be renamed to `lateInput` which is matched with metric name. Probably the variable `predicate` is also not clear which row will get `true`. Let me apply it.